### ISSUE TITLE
Machine requires VM to be running to upgrade, then immediately stops it

### ISSUE THREAD

**Issue by dmlyons** [docker/machine#2228]
When running `docker-machine upgrade` it will require the machine to be running to upgrade, and the first thing it does is stop the machine.  Why can't it perform the upgrade without the machine running?  Shouldn't it do the upgrade without throwing the error?

```
12:59 $ docker-machine upgrade dev
Error: machine must be running to upgrade.
✘-1 ~ dlyons
12:59 $ docker-machine start dev

Started machines may have new IP addresses. You may need to re-run the `docker-machine env` command.
✔ ~ dlyons
12:59 $ docker-machine upgrade dev
Stopping machine to do the upgrade...
```

**Comment by nathanleclaire** [comment]
It's a known issue (https://github.com/docker/machine/issues/1399)

Machine needs the machine to be up in order to ask them which OS they are so it can know how to proceed with the upgrade.

We might store the computed OS information in the future since it's unlikely to change much.

**Comment by dmlyons** [comment]
Ah, thanks, my search-fu failed me, didn't mean to open a duplicate.

**Comment by jeanlaurent** [comment]
Closed as duplicate of #1399 

Feel free to re-open if I missed something

**Comment by jeanlaurent** [comment]
Closed as duplicate of #1399 

Feel free to re-open if I missed something
